rs/replicated_state/src/canister_state.rs

```diff
@@ -144,10 +144,14 @@ impl CanisterState {
         }
     }
 
-    /// Apply priority credit
+    /// Applies priority credit and resets long execution mode.
     pub fn apply_priority_credit(&mut self) {
         self.scheduler_state.accumulated_priority -=
             std::mem::take(&mut self.scheduler_state.priority_credit);
+        // Aborting a long-running execution moves the canister to the
+        // default execution mode because the canister does not have a
+        // pending execution anymore.
+        self.scheduler_state.long_execution_mode = LongExecutionMode::default();
```

Comment:
This is only called from one location, behind an `if !has_aborted_or_paused_execution` check. So resetting `long_execution_mode` is always a no-op. I would instead add a `debug_assert!()` that `long_execution_mode == LongExecutionMode::default()`.

Reply:
It does not look like a no-op to me. We set in the `long_execution_mode` to `Prioritized` in [apply_scheduling_strategy](https://github.com/dfinity/ic/blob/90e12e0aae718198304e8ed42464171e5878183c/rs/execution_environment/src/scheduler.rs#L382), and we reset it here.